packages/swapper/src/swappers/thorchain/ThorchainSwapper.ts

```diff
@@ -133,21 +137,23 @@ export class ThorchainSwapper implements Swapper<ChainId> {
           fn: 'executeTrade'
         })
 
-      if (trade.sellAsset.chainId === KnownChainIds.EthereumMainnet) {
+      const { chainNamespace } = fromAssetId(trade.sellAsset.assetId)
+
+      if (chainNamespace === CHAIN_NAMESPACE.Ethereum) {
         const signedTx = await (adapter as unknown as ethereum.ChainAdapter).signTransaction({
           txToSign: (trade as ThorTrade<KnownChainIds.EthereumMainnet>).txData as ETHSignTx,
           wallet
         })
         const txid = await adapter.broadcastTransaction(signedTx)
         return { tradeId: txid }
-      } else if (trade.sellAsset.chainId === KnownChainIds.BitcoinMainnet) {
+      } else if (chainNamespace === CHAIN_NAMESPACE.Bitcoin) {
         const signedTx = await (adapter as unknown as bitcoin.ChainAdapter).signTransaction({
```

Comment:
Should be updated to cast as a `UtxoBaseAdapter`

Reply:
Same could be said for `EvmBaseAdapter` over `ethereum.ChainAdapter`

Reply:
UtxoBaseAdapter worked but ran into type errors casting to EvmBaseAdapter

Reply:
This worked for me @elmutt, could you try it out?

```diff
diff --git a/packages/swapper/src/swappers/thorchain/ThorchainSwapper.ts b/packages/swapper/src/swappers/thorchain/ThorchainSwapper.ts
index 6fbf8484..0e27a6de 100644
--- a/packages/swapper/src/swappers/thorchain/ThorchainSwapper.ts
+++ b/packages/swapper/src/swappers/thorchain/ThorchainSwapper.ts
@@ -1,6 +1,6 @@
 import { Asset } from '@shapeshiftoss/asset-service'
 import { adapters, AssetId, CHAIN_NAMESPACE, ChainId, fromAssetId } from '@shapeshiftoss/caip'
-import { bitcoin, cosmos, ethereum } from '@shapeshiftoss/chain-adapters'
+import { cosmos, EvmBaseAdapter, UtxoBaseAdapter } from '@shapeshiftoss/chain-adapters'
 import { BTCSignTx, CosmosSignTx, ETHSignTx } from '@shapeshiftoss/hdwallet-core'
 import { KnownChainIds } from '@shapeshiftoss/types'
 
@@ -10,6 +10,7 @@ import {
   ApproveInfiniteInput,
   BuildTradeInput,
   BuyAssetBySellIdInput,
+  EvmSupportedChainIds,
   ExecuteTradeInput,
   GetTradeQuoteInput,
   SwapError,
@@ -142,14 +143,18 @@ export class ThorchainSwapper implements Swapper<ChainId> {
       const { chainNamespace } = fromAssetId(trade.sellAsset.assetId)
 
       if (chainNamespace === CHAIN_NAMESPACE.Ethereum) {
-        const signedTx = await (adapter as unknown as ethereum.ChainAdapter).signTransaction({
+        const signedTx = await (
+          adapter as unknown as EvmBaseAdapter<EvmSupportedChainIds>
+        ).signTransaction({
           txToSign: (trade as ThorTrade<KnownChainIds.EthereumMainnet>).txData as ETHSignTx,
           wallet
         })
         const txid = await adapter.broadcastTransaction(signedTx)
         return { tradeId: txid }
       } else if (chainNamespace === CHAIN_NAMESPACE.Bitcoin) {
-        const signedTx = await (adapter as unknown as bitcoin.ChainAdapter).signTransaction({
+        const signedTx = await (
+          adapter as unknown as UtxoBaseAdapter<UtxoSupportedChainIds>
+        ).signTransaction({
           txToSign: (trade as ThorTrade<UtxoSupportedChainIds>).txData as BTCSignTx,
           wallet
         })
```

Reply:
oh ya that works just fine, thanks! My ide must have lying to me about a type error